apps/sms/js/subject_composer.js

```diff
@@ -106,7 +113,8 @@
 
       // methods
       updateValue: updateValue.bind(this),
-      updatePlaceholder: updatePlaceholder.bind(this)
+      updatePlaceholder: updatePlaceholder.bind(this),
+      updateDirection: updateDirection.bind(this)
```

Comment:
but "input" and "updateDirection" are in the same object. So when calling "priv.updateDirection", `this` is set to `priv` :) That said I'm not sure this is such a good idea after all, for consistency. Let's keep it like this.